internal/git/git.go

```diff
@@ -16,26 +18,64 @@ const (
 
 // copied from goreleaser
 
-// IsRepo returns true if current folder is a git repository
-func IsRepo() bool {
-	out, err := run("rev-parse", "--is-inside-work-tree")
-	return err == nil && strings.TrimSpace(out) == "true"
+type Repository struct {
+	GitWorkTree  string
+	GitDirectory string
 }
 
-func getAllTags(args ...string) ([]string, error) {
-	tags, err := run(append([]string{"-c", "versionsort.suffix=-", "tag", "--sort=-version:refname"}, args...)...)
+// NewRepository creates a Repository. gitworktree and gitdirectory will default to 'pwd' and 'pwd/.git'
+func NewRepository(gitworktree, gitdirectory string) (*Repository, error) {
```

Comment:
It might be overkill when there is only two argument, but an Option pattern here may help a lot for readability.


```suggestion
func NewRepository(opts...RepositoryOption) (*Repository, error) {
```

So we can do

git.NewRepository(git.FromDirectory("whatever"))

git.NewRepository(git.FromWorkspace("whatever"))


I let @caarlos0 share his thoughts